### PR TITLE
Filter out goog_ user timings.

### LINKED ISSUE #1548: filter irrelevant user timing marks
Unclear what basis to do this on (hardcoded?), but google ads, for instance are injecting a bunch of measurements now:

### PR DESCRIPTION
**Pull request by paulirish**
Fixes #1548

### CODE REVIEW

File: lighthouse-core/audits/user-timings.js [hunk 1]
```diff
@@ -125,13 +125,22 @@ class UserTimings extends Audit {
     };
   }
 
+  // We'll remove mark/measures entered by third parties not of interest to the user
+  static get blacklistedPrefixes() {
+    return ['goog_'];
+  }
+
+  static excludeBlacklisted(timing) {
```
Comment (brendankenny) [inline lighthouse-core/audits/user-timings.js]: jsdoc would be good (nit: and maybe comment above should actually go on this function)

Comment (paulirish) [inline lighthouse-core/audits/user-timings.js]: k done

File: lighthouse-core/audits/user-timings.js [hunk 2]
```diff
@@ -126,12 +126,28 @@ class UserTimings extends Audit {
...
+   * @return {!Array<string>}
+   */
+  static get blacklistedPrefixes() {
+    return ['goog_'];
+  }
+
+  /**
+   * We remove mark/measures entered by third parties not of interest to the user
+   * @param {!UserTimingsExtendedInfo} artifacts
+   * @return {boolean>}
```
Comment (brendankenny) [inline lighthouse-core/audits/user-timings.js]: stray `>`

Comment (paulirish) [inline lighthouse-core/audits/user-timings.js]: fixed

**Comment by brendankenny** [comment]
needs a rebase now :)

File: lighthouse-core/audits/user-timings.js [hunk 3]
```diff
@@ -126,12 +126,28 @@ class UserTimings extends Audit {
...
+    return ['goog_'];
+  }
+
+  /**
+   * We remove mark/measures entered by third parties not of interest to the user
+   * @param {!UserTimingsExtendedInfo} artifacts
+   * @return {boolean}
+   */
+  static excludeBlacklisted(timing) {
+    return UserTimings.blacklistedPrefixes.every(prefix => !timing.name.startsWith(prefix));
```
Comment (ebidel) [inline lighthouse-core/audits/user-timings.js]: It's that Node > 4 bling ✨

**Comment by paulirish** [comment]
Conflicts resolved. Merging!

**Pull request merged**